Dockerfile

```diff
@@ -0,0 +1,10 @@
+FROM python:3.12-slim
+
+WORKDIR /app
+
+COPY requirements.txt .
+RUN pip install --no-cache-dir -r requirements.txt
+
+COPY . .
```

Comment:
Since you don't have `.dockerignore` file (just like `.gitignore`, but for Docker), you'll actually copy ALL of your project's files into the . And you want to keep your Docker images and containers slim as possible.